packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,217 +73,375 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
                 ));
             }
         }
-        assert_eq!(fns.len(), 3);
 
         Ok(JsPackage { fns })
     }
 }
 
 /// Embedded JS of runner itself (from hardcoded paths)
-struct Embedded<'m> {
-    hash_stdlib: mv8::Value<'m>,
-    hash_util: mv8::Value<'m>,
-
-    start_experiment: mv8::Function<'m>,
-    start_sim: mv8::Function<'m>,
-    run_task: mv8::Function<'m>,
-    ctx_batch_sync: mv8::Function<'m>,
-    state_sync: mv8::Function<'m>,
-    state_interim_sync: mv8::Function<'m>,
-    state_snapshot_sync: mv8::Function<'m>,
+struct Embedded<'s> {
+    hash_stdlib: v8::Local<'s, v8::Value>,
+    hash_util: v8::Local<'s, v8::Value>,
+
+    start_experiment: v8::Local<'s, v8::Function>,
+    start_sim: v8::Local<'s, v8::Function>,
+    run_task: v8::Local<'s, v8::Function>,
+    ctx_batch_sync: v8::Local<'s, v8::Function>,
+    state_sync: v8::Local<'s, v8::Function>,
+    state_interim_sync: v8::Local<'s, v8::Function>,
+    state_snapshot_sync: v8::Local<'s, v8::Function>,
 }
 
-fn read_file(path: &str) -> Result<String> {
-    fs::read_to_string(path).map_err(|e| Error::IO(path.into(), e))
+fn read_file(path: &str) -> JSRunnerResult<String> {
+    fs::read_to_string(path).map_err(|e| JSRunnerError::IO(path.into(), e))
 }
 
-fn eval_file<'m>(mv8: &'m MiniV8, path: &str) -> Result<mv8::Value<'m>> {
-    let code = read_file(path)?;
-    let v: mv8::Value<'_> = mv8
-        .eval(code)
-        .map_err(|e| Error::Eval(path.into(), e.into()))?;
-    Ok(v)
+fn eval_file<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    path: &str,
+) -> JSRunnerResult<v8::Local<'s, v8::Value>> {
+    let source_code = read_file(path)?;
+    let js_source_code = new_js_string(scope, &source_code)?;
+    let script = v8::Script::compile(scope, js_source_code, None)
+        .ok_or_else(|| JSRunnerError::Eval(path.into(), "compile error".to_string()))?;
+
+    script
+        .run(scope)
+        .ok_or_else(|| JSRunnerError::Eval(path.into(), "execution error".to_string()))
 }
 
-fn import_file<'m>(
-    mv8: &'m MiniV8,
+fn import_file<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    context: v8::Local<'s, v8::Context>,
     path: &str,
-    args: Vec<&mv8::Value<'m>>,
-) -> Result<mv8::Value<'m>> {
-    let v = eval_file(mv8, path)?;
-    let f = v
-        .as_function()
-        .ok_or_else(|| Error::FileImport(path.into(), format!("Failed to wrap file: {:?}", &v)))?;
-
-    let args = {
-        let mut a = Vec::new();
-        for arg in args {
-            a.push(arg.clone());
-        }
-        a
-    };
-    let args = mv8::Values::from_vec(args);
-    let imported = f
-        .call(args)
-        .map_err(|e| Error::FileImport(path.into(), e.into()))?;
+    args: &[v8::Local<'s, v8::Value>],
+) -> JSRunnerResult<v8::Local<'s, v8::Value>> {
+    let v = eval_file(scope, path)?;
+    let f: v8::Local<'_, v8::Function> = v.try_into().map_err(|e| {
+        JSRunnerError::FileImport(path.into(), format!("Failed to wrap file: {:?}, {e}", &v))
```

Comment:
```suggestion
        JSRunnerError::FileImport(path.into(), format!("Failed to wrap file: {v:?}, {e}"))
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/02af15786b564ada4f9992194281ff0a3281afb9